docs/functions.md

```diff
@@ -0,0 +1,567 @@
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withDisableCapacities true ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withContinueOnMethod` <a id="withContinueOnMethod"></a>
+
+Sets the `ContinueOnMethod` flag for a customized Spew printer. From the Spew documentation:
+
+> `ContinueOnMethod` specifies whether or not recursion should continue once
+> a custom error or Stringer interface is invoked.  The default, `false`,
+> means it will print the results of invoking the custom `error` or `Stringer`
+> interface and return immediately instead of continuing to recurse into
+> the internals of the data type.
+>
+> **NOTE:** This flag does not have any effect if method invocation is disabled
+> via the `DisableMethods` or `DisablePointerMethods` options.
+
+##### Parameters
+
+- 1: `bool` - The value to set the `ContinueOnMethod` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withContinueOnMethod true ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withSortKeys` <a id="withSortKeys"></a>
+
+Sets the `SortKeys` flag for a customized Spew printer. From the Spew documentation:
+
+> `SortKeys` specifies map keys should be sorted before being printed. Use
+> this to have a more deterministic, diffable output.  Note that only
+> native types (`bool`, `int`, `uint`, `floats`, `uintptr`, and `string`) and types
+> that support the error or `Stringer` interfaces (if methods are
+> enabled) are supported, with other types sorted according to the
+> `reflect.Value.String()` output which guarantees display stability.
+
+##### Parameters
+
+- 1: `bool` - The value to set the `SortKeys` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withSortKeys true ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withSpewKeys` <a id="withSpewKeys"></a>
+
+Sets the `SpewKeys` flag for a customized Spew printer. From the Spew documentation:
+
+> `SpewKeys` specifies that, as a last resort attempt, map keys should
+> be spewed to strings and sorted by those strings.  This is only
+> considered if `SortKeys` is true.
+
+##### Parameters
+
+- 1: `bool` - The value to set the `SpewKeys` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+Returns:
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withSpewKeys true ) ]][[ $cs.Sdump . ]]
+```
+
+### Helper functions <a id="topicHelpers"></a>
+
+#### `fileContents` <a id="fileContents"></a>
+
+Imports the contents of a file on the local file system into the template at runtime.
```

Comment:
Can it be clarified that the files are relative to the directory where nomad-pack is invoked? I'm pretty sure they aren't relative to the job as that isn't working for me (which is frustrating, because thats where I'd expect them to be).

Reply:
The `fileContents` function is part of the job meta-templating and is run while **building** the job spec itself.  Should you want to import files while processing the jobspec , you should consider the HCL2 [`file`](https://developer.hashicorp.com/nomad/docs/job-specification/hcl2/functions/file/file) function, which would work more like you expect; but requires that the file to include always travel with the job spec.